src/adhocracy_mercator/adhocracy_mercator/scripts/export_users.py

```diff
@@ -7,16 +7,61 @@
 import sys
 import textwrap
 import time
-
 from pyramid.paster import bootstrap
 from substanced.util import find_catalog
 
+from adhocracy_core.catalog.adhocracy import index_rates
+from adhocracy_core.resources.principal import IUser
+from adhocracy_core.sheets.metadata import IMetadata
 from adhocracy_core.sheets.principal import IUserBasic
 from adhocracy_core.sheets.principal import IUserExtended
-from adhocracy_core.sheets.metadata import IMetadata
+from adhocracy_core.sheets.rate import IRateable
+from adhocracy_core.sheets.rate import IRate
 from adhocracy_core.utils import get_sheet_field
+from adhocracy_mercator.resources.mercator import IMercatorProposalVersion
+from adhocracy_mercator.sheets.mercator import ITitle
+from adhocracy_core.sheets.tags import filter_by_tag
+
+
+def _get_most_rated_proposals(root, min_rate=100):
+    catalog = find_catalog(root, 'system')
+    adhocracy_catalog = find_catalog(root, 'adhocracy')
+    path = catalog['path']
+    interfaces = catalog['interfaces']
+    tag = adhocracy_catalog['tag']
+
+    query = path.eq('/mercator') \
+        & interfaces.eq(IMercatorProposalVersion) \
+        & tag.eq('LAST')
+
+    proposals = query.execute()
+    proposals = [p for p in proposals if index_rates(p, None) > min_rate]
+    proposals = list(reversed(sorted(proposals,
+                                     key=lambda p: index_rates(p, None))))
+    return proposals
 
-from adhocracy_core.resources.principal import IUser
+
+def _get_proposals_titles(proposals):
+    return [get_sheet_field(p, ITitle, 'title') for p in proposals]
+
+
+def _get_rate_subject_name(rate):
+    rater = get_sheet_field(rate, IRate, 'subject')
+    return get_sheet_field(rater, IUserBasic, 'name')
+
+
+def _get_user_rate(user_name, proposal):
+    rates = get_sheet_field(proposal, IRateable, 'rates')
+    last_rates = filter_by_tag(rates, 'LAST')
+    rated = [rate for rate in last_rates
+             if _get_rate_subject_name(rate) == user_name]
+
+    if len(rated) == 1:
+        rate = rated[0]
+        creation_date = get_sheet_field(rate, IMetadata, 'item_creation_date')
+        return (get_sheet_field(rate, IRate, 'rate'),
+                creation_date.strftime('%Y-%m-%d_%H:%M:%S'))
+    return (0, '')
 
 
 def export_users():
```

Comment:
To  export all user it would be much simpler to just use bin/sd_dump   ../principals/users
But in addition this function add most  rated proposals. This is not clear when reading the function name or docstring.